### PR TITLE
JENA-2068: Fuseki main jetty and port configuration

### PR DESCRIPTION
**Pull request by afs**
Includes handling port 0 - allocate atomically when the server starts then finding what we got - in place of the "ask" then use the given port which has a timing hole albeit small.

### CODE REVIEW

File: jena-fuseki2/jena-fuseki-main/src/main/java/org/apache/jena/fuseki/main/cmds/FusekiMain.java [hunk 1]
```diff
@@ -414,13 +433,14 @@ protected void processModulesAndArgs() {
 
     private int portNumber(ArgDecl arg) {
         String portStr = getValue(arg);
+        if ( portStr.isEmpty() )
```
Comment (kinow) [inline jena-fuseki2/jena-fuseki-main/src/main/java/org/apache/jena/fuseki/main/cmds/FusekiMain.java:436]: Not sure if blanks could also be handled here with `StringUtils.isBlank` or .trim()?

File: jena-fuseki2/jena-fuseki-main/src/test/java/org/apache/jena/fuseki/main/TestFusekiServerBuild.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.apache.jena.rdfconnection.RDFConnection;
+import org.apache.jena.rdfconnection.RDFConnectionFactory;
+import org.apache.jena.sparql.core.DatasetGraphFactory;
+import org.junit.Test;
+
+public class TestFusekiServerBuild {
+    // Most testing happens by use in the test suit.
```
Comment (kinow) [inline jena-fuseki2/jena-fuseki-main/src/test/java/org/apache/jena/fuseki/main/TestFusekiServerBuild.java:31]: s/suit/suite

**Review by kinow — APPROVED** [review]

File: jena-fuseki2/jena-fuseki-main/src/main/java/org/apache/jena/fuseki/main/cmds/FusekiMain.java (hunk 1 above)
Comment (afs) [reply jena-fuseki2/jena-fuseki-main/src/main/java/org/apache/jena/fuseki/main/cmds/FusekiMain.java:436]: This is from the command line which will have discarded (unquoted) whitespace.

`--port=""` or `--port=` is clearer whitespace.

There is a argument for applying it to numbers but the command hasn't up to now.

File: jena-fuseki2/jena-fuseki-main/src/test/java/org/apache/jena/fuseki/main/TestFusekiServerBuild.java (hunk 2 above)
Comment (afs) [reply jena-fuseki2/jena-fuseki-main/src/test/java/org/apache/jena/fuseki/main/TestFusekiServerBuild.java:31]: Done - thanks.

**Pull request merged**